### PR TITLE
Add example of how to plot a 3d vector field

### PR DESCRIPTION
**Pull request by LSchueler**
In this short example I demonstrate how to use PyVista to create an external mesh on which a 3d vector field is generated.
I'm not sure why the colours look a but dull in Sphinx-Gallery. When looking at the plot in PyVista, the colours are more vibrant. But this is not a big issue I guess.

### CODE REVIEW

File: examples/04_vector_field/01_3d_vector_field.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+In this example we are going to generate a random 3D vector field with a
+Gaussian covariance model. The mesh on which we generate the field will be
+externally defined and it will be generated by PyVista.
+"""
+import numpy as np
+import gstools as gs
+import pyvista as pv
+
+pv.set_plot_theme("document")
```
Comment (MuellerSeb) [inline examples/04_vector_field/01_3d_vector_field.py:13]: Why this?

File: examples/04_vector_field/01_3d_vector_field.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+###############################################################################
+# create a uniform grid with PyVista
+nx, ny, nz = 40, 30, 10
+mesh = pv.UniformGrid((nx, ny, nz), (1., 1., 1.), (-10., 0., 0.))
```
Comment (MuellerSeb) [inline examples/04_vector_field/01_3d_vector_field.py:18]: would prefere:
```python
dim, spacing, origin = (40, 30, 10), (1, 1, 1), (-10, 0, 0)
mesh = pv.UniformGrid(dim, spacing, origin)
```
Following the documentation of pyvista.

File: examples/04_vector_field/01_3d_vector_field.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+model = gs.Gaussian(dim=3, var=3, len_scale=1.5)
+srf = gs.SRF(model, generator='VectorField')
+srf((x, y, z), mesh_type='unstructured', seed=198412031)
+
+# add a mean velocity in x-direction
+srf.field[0,:] += 0.5
+
+###############################################################################
+# add the velocity field to the mesh object
+mesh["Velocity"] = srf.field.T
```
Comment (MuellerSeb) [inline examples/04_vector_field/01_3d_vector_field.py:34]: Why not using the `mesh` method? And set seed in `__init__` so sampling is done only once,
```python
srf = gs.SRF(model, generator='VectorField', seed=198412031)
srf.mesh(mesh, points="points", name="Velocity")
mesh["Velocity"][:, 0] += 0.5
```

File: examples/04_vector_field/01_3d_vector_field.py [hunk 4]
```diff
@@ -0,0 +1,70 @@
+"""
+Generating a Random 3D Vector Field
+-----------------------------------
+
+In this example we are going to generate a random 3D vector field with a
+Gaussian covariance model. The mesh on which we generate the field will be
+externally defined and it will be generated by PyVista.
+"""
+import numpy as np
```
Comment (MuellerSeb) [inline examples/04_vector_field/01_3d_vector_field.py:9]: missing thumbnail in gallery:
```python
# sphinx_gallery_thumbnail_path = 'pics/GS_pyvista_streamlines.png'
```

**Review by MuellerSeb — CHANGES_REQUESTED** [review]
That's awesome! Some minor things from my side in the review.
And: I would rename `mesh` to `grid` following the pyvista documentation on `UniformGrid`.
Also the thumbnail is missing. Have a look at the other pyvista example for this.

File: examples/04_vector_field/01_3d_vector_field.py (hunk 1 above)
Comment (LSchueler) [reply examples/04_vector_field/01_3d_vector_field.py:13]: clarified in example

File: examples/04_vector_field/01_3d_vector_field.py (hunk 4 above)
Comment (LSchueler) [reply examples/04_vector_field/01_3d_vector_field.py:9]: Good catch, thanks!

File: examples/04_vector_field/01_3d_vector_field.py (hunk 2 above)
Comment (LSchueler) [reply examples/04_vector_field/01_3d_vector_field.py:18]: Yeah, that makes things clearer.

**Comment by LSchueler** [comment]
> That's awesome! Some minor things from my side in the review.
> And: I would rename `mesh` to `grid` following the pyvista documentation on `UniformGrid`.
> Also the thumbnail is missing. Have a look at the other pyvista example for this.

Yeah, that seems to be a bit inconsistent in PyVista, I followed the example [here](https://docs.pyvista.org/examples/01-filter/streamlines.html#custom-3d-vector-field) and I call the method `Plotter.add_mesh()` in this example, so I think it's the least confusing if I stick to `mesh`. And `SRF` also has a `mesh()` method ;-) .

File: examples/04_vector_field/01_3d_vector_field.py (hunk 3 above)
Comment (LSchueler) [reply examples/04_vector_field/01_3d_vector_field.py:34]: :-D I'm simply not keeping up with the development of GSTools... But I think the method is not working with vector fields... I'm gonna test that again and open an issue if that is true.
I suggest to keep the example as it is for the moment and update it, as soon as I figured out the problem.

Comment (MuellerSeb) [reply examples/04_vector_field/01_3d_vector_field.py]: You are right... it's not working with Vector fields. Damn.. please create an issue.

File: examples/04_vector_field/01_3d_vector_field.py (hunk 1 above)
Comment (MuellerSeb) [reply examples/04_vector_field/01_3d_vector_field.py:15]: Cool!

File: examples/04_vector_field/01_3d_vector_field.py [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+z = mesh.points[:, 2]
+
+###############################################################################
+# create an incompressible random 3d velocity field on the given mesh
+model = gs.Gaussian(dim=3, var=3, len_scale=1.5)
+srf = gs.SRF(model, generator='VectorField', seed=198412031)
+srf((x, y, z), mesh_type='unstructured')
+
+# add a mean velocity in x-direction
+srf.field[0, :] += 0.5
```
Comment (MuellerSeb) [inline examples/04_vector_field/01_3d_vector_field.py:33]: Couldn't this be provided by a mean? I think the new mean-normalizer-trend thingy is capable of dealing with vectors. But this is just fine here ;-)

**Review by MuellerSeb — APPROVED** [review]
I fine with this fine example :relaxed:

File: examples/04_vector_field/01_3d_vector_field.py (hunk 3 above)
Comment (LSchueler) [reply examples/04_vector_field/01_3d_vector_field.py]: Will do, but I won't make it today anymore...

File: examples/04_vector_field/01_3d_vector_field.py (hunk 5 above)
Comment (LSchueler) [reply examples/04_vector_field/01_3d_vector_field.py:33]: I'll update the example when the normalizer is ready. This example is gonna show a lot of fancy things ;-)

**Comment by MuellerSeb** [comment]
Info:
vector valued mean added: https://github.com/GeoStat-Framework/GSTools/pull/124/commits/9bb8daafc08880eac38a15c3b86a0e540bcf9785
mesh call for vector valued fields fixed: https://github.com/GeoStat-Framework/GSTools/pull/124/commits/56ec497e2d52079c55f74ff8012d0fa9876e7661

**Pull request merged**